x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/formula/formula.tsx

```diff
@@ -0,0 +1,881 @@
+  params: {
+    formula?: string;
+    isFormulaBroken?: boolean;
+    // last value on numeric fields can be formatted
+    format?: {
+      id: string;
+      params?: {
+        decimals: number;
+      };
+    };
+  };
+}
+
+export const formulaOperation: OperationDefinition<
+  FormulaIndexPatternColumn,
+  'managedReference'
+> = {
+  type: 'formula',
+  displayName: defaultLabel,
+  getDefaultLabel: (column, indexPattern) => defaultLabel,
+  input: 'managedReference',
+  hidden: true,
+  getDisabledStatus(indexPattern: IndexPattern) {
+    return undefined;
+  },
+  getErrorMessage(layer, columnId, indexPattern, operationDefinitionMap) {
+    const column = layer.columns[columnId] as FormulaIndexPatternColumn;
+    if (!column.params.formula || !operationDefinitionMap) {
+      return;
+    }
+    const { root, error } = tryToParse(column.params.formula);
+    if (error || !root) {
+      return [error!.message];
+    }
+
+    const errors = runASTValidation(root, layer, indexPattern, operationDefinitionMap);
+    return errors.length ? errors.map(({ message }) => message) : undefined;
+  },
+  getPossibleOperation() {
+    return {
+      dataType: 'number',
+      isBucketed: false,
+      scale: 'ratio',
+    };
+  },
+  toExpression: (layer, columnId) => {
+    const currentColumn = layer.columns[columnId] as FormulaIndexPatternColumn;
+    const params = currentColumn.params;
+    // TODO: improve this logic
+    const useDisplayLabel = currentColumn.label !== defaultLabel;
+    const label = !params?.isFormulaBroken
+      ? useDisplayLabel
+        ? currentColumn.label
+        : params?.formula
+      : '';
+
+    return currentColumn.references.length
+      ? [
+          {
+            type: 'function',
+            function: 'mapColumn',
+            arguments: {
+              id: [columnId],
+              name: [label || ''],
+              exp: [
+                {
+                  type: 'expression',
+                  chain: [
+                    {
+                      type: 'function',
+                      function: 'math',
+                      arguments: {
+                        expression: [`${currentColumn.references[0]}`],
+                      },
+                    },
+                  ],
+                },
+              ],
+            },
+          },
+        ]
+      : [];
+  },
+  buildColumn({ previousColumn, layer }, _, operationDefinitionMap) {
+    let previousFormula = '';
+    if (previousColumn) {
+      if ('references' in previousColumn) {
+        const metric = layer.columns[previousColumn.references[0]];
+        if (metric && 'sourceField' in metric) {
+          const fieldName = getSafeFieldName(metric.sourceField);
+          // TODO need to check the input type from the definition
+          previousFormula += `${previousColumn.operationType}(${metric.operationType}(${fieldName})`;
+        }
+      } else {
+        if (previousColumn && 'sourceField' in previousColumn) {
```

Comment:
Do we have to check whether the previous column is number based here? It might not in case of last value